src/NosCore.GameObject/ComponentEntities/Extensions/AliveEntityExtension.cs

```diff
@@ -270,20 +282,28 @@ public static void SetLevel(this INamedEntity experiencedEntity, byte level)
             byte shopKind)
         {
             var shopItemList = new List<NInvItemSubPacket>();
-            foreach (var item in aliveEntity.Shop.ShopItems.Values.Where(s => s.Type.Equals(typeshop))
-                .OrderBy(s => s.Slot))
+            var list = aliveEntity.Shop.ShopItems.Values.Where(s => s.Type.Equals(typeshop)).ToList();
+            for (var i = 0; i < aliveEntity.Shop.Size; i++)
             {
-                shopItemList.Add(new NInvItemSubPacket
+                var item = list.Find(s=>s.Slot == i);
+                if (item == null)
+                {
+                    shopItemList.Add(null);
+                }
```

Comment:
this feels weird, is it normal ?

Reply:
yes it is. we want the item to be null to get -1